src/net_processing.cpp

```diff
@@ -2575,6 +2575,8 @@ bool SendMessages(CNode* pto, CConnman& connman)
             state.fShouldBan = false;
             if (pto->fWhitelisted)
                 LogPrintf("Warning: not punishing whitelisted peer %s!\n", pto->addr.ToString());
+            else if (pto->fAddnode)
```

Comment:
"Ban-disconnects are also exempted for peers added via addnode since
the outbound addnode logic ignores bans. Previously it would ban
an addnode then immediately reconnect to it."